hack/gcp-service-account-maintenance/build-service-account.bash

```diff
@@ -0,0 +1,89 @@
+GCLOUD_SA_EMAIL="${GCLOUD_SA}@${GCLOUD_PROJECT}.iam.gserviceaccount.com"
+
+GCLOUD="gcloud --project ${GCLOUD_PROJECT}"
+
+echo
+echo "checking connection:"
+CURRENT_OC_CONTEXT=`oc config view -o json | jq -r '.["current-context"]'`
+CURRENT_OC_SERVER=`oc config view -o json | jq -r ".contexts[] | select(.name == \"${CURRENT_OC_CONTEXT}\") | .context.cluster"`
+if [ ! "${CURRENT_OC_SERVER}" == "${TARGET_OC_SERVER}" ]; then
+    echo "oc cli configured to incorrect context, expected [${TARGET_OC_SERVER}] got [${CURRENT_OC_SERVER}]";
+    echo "please log into your expected cluster and try again, or reconfigure ./config.source";
+    exit 1;
+fi
+echo "connected to [${CURRENT_OC_SERVER}]"
+
+echo
+echo "checking ${GCLOUD}:"
+GCLOUD_USER=`${GCLOUD} config get-value core/account`
+if [ ! "${GCLOUD_USER}" == "${TARGET_GCLOUD_USER}" ]; then
+    echo "oc cli configured to incorrect context, expected [${TARGET_GCLOUD_USER}] got [${GCLOUD_USER}]";
+    echo "please log into your expected gcloud account and try again, or reconfigure ./config.source";
+    exit 1;
+fi
+echo "connected to ${GCLOUD} with [${GCLOUD_USER}]"
+
+
+echo
+echo "checking existing service account [${GCLOUD_SA}]"
+${GCLOUD} iam service-accounts list --quiet --filter name:"${GCLOUD_SA}" | grep "${GCLOUD_SA}"
+[ $? == 0 ] && { SA_EXISTS=1; } || SA_EXISTS=0
+
+if [ ! ${SA_EXISTS} ] ; then
+    echo "creating service account [${GCLOUD_SA}]"
+    ## TODO: Change URL to being where this script is located
+    ${GCLOUD} iam service-accounts create ${GCLOUD_SA} \
+        --description="date: ${DATE}; docs: https://github.com/openshift/gcp-project-operator/blob/master/docs/gcpconfig.md" \
+        --display-name="${GCLOUD_SA}"
+fi
+
+echo
+echo "enable service account [" ${GCLOUD_SA} "]"
+${GCLOUD} iam service-accounts enable ${GCLOUD_SA_EMAIL}
+
+echo
+echo "set service account roles [${GCLOUD_SA}]"
+
+# Required Roles are defined here:
+# https://github.com/openshift/gcp-project-operator/blob/master/pkg/controller/projectreference/projectreference_adapter.go#L58-L68
+for required_role in \
+    roles/storage.admin \
+    roles/iam.serviceAccountUser \
+    roles/iam.serviceAccountKeyAdmin \
+    roles/iam.serviceAccountAdmin \
+    roles/iam.securityAdmin \
+    roles/dns.admin \
+    roles/compute.admin \
+; do
+    ${GCLOUD} projects add-iam-policy-binding ${GCLOUD_PROJECT} \
+        --member=serviceAccount:${GCLOUD_SA_EMAIL} --role=${required_role}
+done
+
+echo
+echo "check existing service account keys [${GCLOUD_SA}]"
+${GCLOUD} iam service-accounts keys list --iam-account ${GCLOUD_SA_EMAIL}
+
+echo
+echo "downloading service account keys [${GCLOUD_SA}]"
+${GCLOUD} iam service-accounts keys create ./key.json --iam-account ${GCLOUD_SA_EMAIL}
+
+echo
+echo "Note: After you create a key, you might need to wait for 60 seconds or more before you use the key. If you try to use a key immediately after you create it, and you receive an error, wait at least 60 seconds and try again."
```

Comment:
Try to use $ echo -e "\nNote: After you create a key, you might need to wait for 60 seconds or more before you use the key. If you try to use a key immediately after you create it, and you receive an error, wait at least 60 seconds and try again." instead of having multiple echo commands.